src/coroutine/waker.rs

```diff
@@ -25,10 +25,13 @@ impl AsyncioWaker {
         self.0.take();
     }
 
-    pub(super) fn initialize_future<'a>(&'a self, py: Python<'a>) -> PyResult<Option<&'a PyAny>> {
+    pub(super) fn initialize_future<'a>(
+        &'a self,
```

Comment:
I think we can remove the lifetime bound on `self` here. Also renaming `'a` -> `'py` for consistency would be nice.